### DISCUSSION TITLE
Docker layer cache in the autoscaling runner scale sets mode

### DISCUSSION THREAD

**Discussion by serhij-matvejev** [actions/actions-runner-controller#2869] [Questions]
We are using summerwind implementation of ARC and we would like to migrate to new autoscaling runner scale sets mode.

We are using PersistentVolumes to implement docker layer cache like described [here](https://github.com/actions/actions-runner-controller/blob/3ede9b5a0159a5e0703ccae6eebfdc89defe2b8f/docs/using-custom-volumes.md#docker-image-layers-caching). The docker image layer caching works almost fine on the summerwind controller on our GKE cluster.

**Problems:**
In new ARC (autoscaling runner scale sets mode) we faced several problems to implement similar solution:
1. Minor: Helm chart does not support extra volumeMounts for dind sidecar in AutoscalingRunnerSet.
2. Major: New ARC leave the PersistentVolume in Released state (instead of Available). So the volume with cache could not be picked up by following runner replicas. One of summerwind's controllers deletes claimRef from the PVs to make them Available ([source code](https://github.com/actions/actions-runner-controller/blob/eaa451df321e3f4708c3ca08ffdb049887ddad9e/controllers/actions.summerwind.net/sync_volumes.go#L172)).

It is possible to clean claimRef in the Released PVs by creating a cronjob, or create some standalone pod with a loop checking or write some mutating webhook to try to deal with it. However this is not nice and in case of cronjob or a pod, it will have larger latency.

**Questions:**
1. Is similar solution (deal with claimRefs) planned in autoscaling runner scale sets mode?
2. Are there any better alternatives to cache the docker layers/images? I don't believe this is rare case so community definitely knows good alternatives.

**Comment by ui-kane-lin** [comment]
Up
